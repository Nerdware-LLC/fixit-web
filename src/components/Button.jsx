import React from "react";
import MaterialButton from "@material-ui/core/Button";
import styled from "styled-components";
import { func, string, element } from "../types";

export const Button = ({ label, children, ...props }) => (
  <StyledButton key={`Button:${label}`} {...props}>
    {label ?? children ?? null}
  </StyledButton>
);

const StyledButton = styled(MaterialButton)`
  box-sizing: border-box;
  font-size: 1rem;
  vertical-align: middle;
  text-decoration: none;
  font-family: Roboto, sans-serif;
  &:hover {
    cursor: pointer;
  }
`;

Button.propTypes = {
  onClick: func,
  label: string,
  children: element
};
