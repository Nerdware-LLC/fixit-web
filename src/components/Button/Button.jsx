import React from "react";
import MuiButton from "@mui/material/Button";
import styled from "@emotion/styled";
import { func, string, element } from "../../types";

// Docs: https://mui.com/components/buttons/

export const Button = ({ label, children, ...props }) => (
  <StyledButton key={`Button:${label}`} {...props}>
    {label ?? children ?? null}
  </StyledButton>
);

const StyledButton = styled(MuiButton)`
  box-sizing: border-box;
  font-size: 1rem;
  vertical-align: middle;
  text-decoration: none;
  font-family: Roboto, sans-serif;
  &:hover {
    cursor: pointer;
  }
`;

Button.propTypes = {
  onClick: func,
  label: string,
  children: element
};
