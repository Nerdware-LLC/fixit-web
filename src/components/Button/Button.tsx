import MuiButton from "@mui/material/Button";
import styled from "@emotion/styled";
import { func, string, element } from "@types";

// Docs: https://mui.com/components/buttons/

export const Button = ({
  label,
  onClick,
  variant = "contained",
  style = {},
  children,
  ...props
}: React.ComponentProps<typeof MuiButton> & { label: string }) => {
  return (
    <StyledButton
      key={`Button:${label}`}
      onClick={onClick}
      variant={variant}
      style={style}
      {...props}
    >
      {label ?? children ?? null}
    </StyledButton>
  );
};

const StyledButton = styled(MuiButton)`
  box-sizing: border-box;
  font-size: 1rem;
  line-height: 1rem;
  vertical-align: middle;
  text-decoration: none;
  font-family: Roboto, sans-serif;
  font-weight: bold;
  &:hover {
    cursor: pointer;
  }
`;

Button.propTypes = {
  label: string,
  onClick: func,
  children: element
};
