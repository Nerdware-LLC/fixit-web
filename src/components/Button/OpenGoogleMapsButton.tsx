import Tooltip from "@mui/material/Tooltip";
import MapMarkerIcon from "@mui/icons-material/Room";
import type { Location } from "@types";

/**
 * Button which will open a given `location` in Google Maps.
 */
export const OpenGoogleMapsButton = ({
  location,
  style = {},
  ...props
}: {
  location: Location;
  style?: React.CSSProperties;
}) => {
  const formattedAddressStr = Object.values(location)
    .filter((locationValue) => typeof locationValue === "string")
    .join(", ");

  const urlEncodedLocation = encodeURIComponent(formattedAddressStr);

  return (
    <Tooltip title={`Google Maps: ${formattedAddressStr}`}>
      <a
        href={`https://www.google.com/maps/dir/?api=1&destination=${urlEncodedLocation}`}
        target="_blank"
        rel="noreferrer"
        {...props}
      >
        <MapMarkerIcon color="primary" />
      </a>
    </Tooltip>
  );
};
