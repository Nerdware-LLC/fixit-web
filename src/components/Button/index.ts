export { Button } from "./Button";
export { OpenGoogleMapsButton } from "./OpenGoogleMapsButton";
