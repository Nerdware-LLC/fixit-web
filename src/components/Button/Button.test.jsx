import renderer from "react-test-renderer";
import { Button } from "./Button";

it("renders correctly", () => {
  const tree = renderer.create(<Button label="Test Button Label" />).toJSON();
  expect(tree).toMatchSnapshot();
});
