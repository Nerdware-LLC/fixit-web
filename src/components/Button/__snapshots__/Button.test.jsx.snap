// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-1p1iq82-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Test Button Label
</button>
`;
